Source/appfat.cpp

```diff
@@ -182,125 +69,59 @@ void assert_fail(int nLineNo, const char *pszFile, const char *pszFail)
 }
 #endif
 
-void DDErrMsg(DWORD error_code, int log_line_nr, char *log_file_path)
-{
-	char *msg;
-
-	if (error_code) {
-		msg = GetErrorStr(error_code);
-		app_fatal("Direct draw error (%s:%d)\n%s", log_file_path, log_line_nr, msg);
-	}
-}
-
-void DSErrMsg(DWORD error_code, int log_line_nr, char *log_file_path)
-{
-	char *msg;
-
-	if (error_code) {
-		msg = GetErrorStr(error_code);
-		app_fatal("Direct sound error (%s:%d)\n%s", log_file_path, log_line_nr, msg);
-	}
-}
-
-void center_window(HWND hDlg)
-{
-	LONG w, h;
-	int screenW, screenH;
-	struct tagRECT Rect;
-	HDC hdc;
-
-	GetWindowRect(hDlg, &Rect);
-	w = Rect.right - Rect.left;
-	h = Rect.bottom - Rect.top;
-	screenW = GetDeviceCaps(hdc, HORZRES);
-	screenH = GetDeviceCaps(hdc, VERTRES);
-
-	if (!SetWindowPos(hDlg, HWND_TOP, (screenW - w) / 2, (screenH - h) / 2, 0, 0, SWP_NOZORDER | SWP_NOSIZE)) {
-		app_fatal("center_window: %s", TraceLastError());
-	}
-}
-
-void ErrDlg(int template_id, DWORD error_code, char *log_file_path, int log_line_nr)
+void ErrDlg(char *title, char *error, char *log_file_path, int log_line_nr)
 {
-	char *size;
-	LPARAM dwInitParam[128];
+	char text[1024];
 
 	FreeDlg();
 
-	size = strrchr(log_file_path, '\\');
-	if (size)
-		log_file_path = size + 1;
-
-	wsprintf((LPSTR)dwInitParam, "%s\nat: %s line %d", GetErrorStr(error_code), log_file_path, log_line_nr);
-	if (DialogBoxParam(ghInst, MAKEINTRESOURCE(template_id), ghMainWnd, (DLGPROC)FuncDlg, (LPARAM)dwInitParam) == -1)
-		app_fatal("ErrDlg: %d", template_id);
+	snprintf(text, 1024, "%s\n\nThe error occurred at: %s line %d", error, log_file_path, log_line_nr);
 
+	UiErrorOkDialog(title, text);
 	app_fatal(NULL);
 }
 
-BOOL __stdcall FuncDlg(HWND hDlg, UINT uMsg, WPARAM wParam, char *text)
-{
-	switch (uMsg) {
-	case WM_INITDIALOG:
-		TextDlg(hDlg, text);
-		break;
-	case WM_COMMAND:
-		if ((WORD)wParam == 1) {
-			EndDialog(hDlg, 1);
-		} else if ((WORD)wParam == 2) {
-			EndDialog(hDlg, 0);
-		}
-		break;
-	default:
-		return FALSE;
-	}
-
-	return TRUE;
-}
-
-void TextDlg(HWND hDlg, char *text)
-{
-	center_window(hDlg);
-
-	if (text)
-		SetDlgItemText(hDlg, 1000, text);
-}
-
 void FileErrDlg(const char *error)
 {
+	char text[1024];
+
 	FreeDlg();
 
 	if (!error)
 		error = "";
-
-	if (DialogBoxParam(ghInst, MAKEINTRESOURCE(IDD_DIALOG3), ghMainWnd, (DLGPROC)FuncDlg, (LPARAM)error) == -1)
-		app_fatal("FileErrDlg");
-
+	snprintf(
+	    text,
+	    1024,
+	    "Unable to open a required file.\n"
+	    "\n"
+	    "Verify that the MD5 of diabdat.mpq matches on of the following values\n"
+	    "011bc6518e6166206231080a4440b373\n"
+	    "68f049866b44688a7af65ba766bef75a\n"
+	    "\n"
+	    "The problem occurred when loading:\n%s",
+	    error);
+
+	UiErrorOkDialog("Data File Error", text);
 	app_fatal(NULL);
 }
 
 BOOL InsertCDDlg()

```

Comment:
The windows build is failing with:
> c:\projects\devilutionx\source\appfat.cpp(116): error C4716: 'dvl::InsertCDDlg': must return a value [C:\projects\devilutionx\devilution.vcxproj]

Probably need to change the type to `void`

Reply:
Turns out this was being in a loop to let the user try to insert the CD up to 5 files, but the way we had stubbed some function made this mostly none functional. I cleaned up the related code so it's more readable and works in a way that I think makes more sense for us (show one error and quit).